server/core/db/boltdb.go

```diff
@@ -106,23 +114,80 @@ func (b *BoltDB) TryLock(newLock models.ProjectLock) (bool, models.ProjectLock,
 		if err := json.Unmarshal(currLockSerialized, &currLock); err != nil {
 			return errors.Wrap(err, "failed to deserialize current lock")
 		}
+
+		// checking if current lock is with the same PR
+		if currLock.Pull.Num == newLock.Pull.Num {
+			lockAcquired = true
+			return nil
+		}
+
 		lockAcquired = false
+
+		// Enqueuing
+		queueBucket := tx.Bucket(b.queueBucketName)
+		currQueueSerialized := queueBucket.Get([]byte(key))
+		// Queue doesn't exist, create one
+		if currQueueSerialized == nil {
+			newQueue := models.ProjectLockQueue{newLock}
```

Comment:
Could you move lines 131 to 142 to its own function and call it here? That way the logic can be more easily unit tested

Reply:
Moved as part of 6937973d90fb5c94a400a1783e8e679966916cf4